pkg/resources/grant_ownership_test.go

```diff
@@ -429,47 +455,33 @@ func TestGetOwnershipGrantTo(t *testing.T) {
 				DatabaseRoleName: sdk.Pointer(sdk.NewDatabaseObjectIdentifier("test_database", "database_role_name")),
 			},
 		},
-		{
-			Name:        "validation - incorrect account role name",
-			AccountRole: sdk.String("database_name.account_role_name"),
-			ExpectPanic: true,
-		},
-		{
-			Name:         "validation - incorrect database role name",
-			DatabaseRole: sdk.String("database_name.schema_name.database_role_name"),
-			ExpectPanic:  true,
-		},
 	}
 
 	for _, tt := range testCases {
 		tt := tt
 		t.Run(tt.Name, func(t *testing.T) {
-			config := make(map[string]any)
-			if tt.AccountRole != nil {
-				config["account_role_name"] = *tt.AccountRole
-			}
-			if tt.DatabaseRole != nil {
-				config["database_role_name"] = *tt.DatabaseRole
-			}
-			d := schema.TestResourceDataRaw(t, grantOwnershipSchema, config)
-
-			defer func() {
-				if err := recover(); err != nil {
-					assert.True(t, tt.ExpectPanic)
-				}
-			}()
-			grantTo := getOwnershipGrantTo(d)
+			grantTo := getOwnershipGrantTo(schema.TestResourceDataRaw(t, grantOwnershipSchema, map[string]any{
+				"database_role_name": *tt.DatabaseRole,
+			}))
 
-			if tt.Expected.AccountRoleName != nil {
-				assert.Equal(t, *tt.Expected.AccountRoleName, *grantTo.AccountRoleName)
-			}
-			if tt.Expected.DatabaseRoleName != nil {
-				assert.Equal(t, *tt.Expected.DatabaseRoleName, *grantTo.DatabaseRoleName)
-			}
+			assert.Equal(t, *tt.Expected.DatabaseRoleName, *grantTo.DatabaseRoleName)
 		})
 	}
 }
 
+func TestInvalidDatabaseRoleGetOwnershipGrantTo(t *testing.T) {
+	d := schema.TestResourceDataRaw(t, grantOwnershipSchema, map[string]any{
+		"database_role_name": "account_role_name",
+	})
+
+	defer func() {
```

Comment:
could be done with assert.Panics as suggested, you can change in the follow-up PR.

Reply:
Changed